### PR TITLE
feat: add ref to ActionPanel

### LINKED ISSUE #962: Allow ActionPanel to have ref
Allow ActionPanel to accept `ref`.

## Types of issue

- [ ] Bug (High impact or critical to a production build)
- [ ] Bug (Low impact or not working as expected)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Something else (idea, build tool, documentation improvement)

## Description

Sometimes we want popovers inside ActionPanel to flip on edges. This is currently done by passing the `ref` to the inner element.
```jsx
<ActionPanel isModal>
  <div ref={someRef}>...</div>
</ActionPanel>
```

However, this brings a limitation that the popover will flip on body boundary, not on the modal boundary.

This issue proposes adding ref to the top level, and attach it to the top level div.
```jsx
// example usage
<ActionPanel ref={someRef}>...</ActionPanel>
```

### PR DESCRIPTION
**Pull request by Harris-YWang**
## Description

Provide a ref prop to the top level for ActionPanel Component so that Devs could better use this component.

Resolves #962 

Also, refactor this component a bit and modify corresponding test cases.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/ActionPanel/index.jsx [hunk 1]
```diff
@@ -1,49 +1,52 @@
...
-  componentWillUnmount() {
-    if (this.props.isModal) document.body.classList.remove('modal-open');
-  }
+  const addBodyClass = classname => document.body.classList.add(classname);
+  const removeBodyClass = classname => document.body.classList.remove(classname);
 
-  render() {
-    const { title, className, size, onClose, children, actionButton, isModal, closeIcon } = this.props;
+  useEffect(() => {
+    // Set up
```
Comment (veegandhi) [inline src/components/ActionPanel/index.jsx]: You can remove these comments.

File: src/components/ActionPanel/index.jsx [hunk 2]
```diff
@@ -55,6 +58,7 @@ ActionPanel.propTypes = {
   actionButton: PropTypes.node,
   closeIcon: PropTypes.node,
   isModal: PropTypes.bool,
+  actionPanelRef: PropTypes.func,
```
Comment (veegandhi) [inline src/components/ActionPanel/index.jsx]: just `ref`

Comment (lightbringer1991) [reply src/components/ActionPanel/index.jsx]: I think the consensus is using `React.forwardRef` to pass on the ref

Comment (Harris-YWang) [reply src/components/ActionPanel/index.jsx]: Solved. `React.forwardRef` does have some new advantages, although it wasn't a breaking update comparing to a custom prop `ref`.

File: src/components/ActionPanel/index.jsx [hunk 3]
```diff
@@ -1,49 +1,50 @@
...
+import React, { useEffect } from 'react';
 import Button from 'react-bootstrap/lib/Button';
 import ReactDOM from 'react-dom';
 import './styles.scss';
 
-class ActionPanel extends React.PureComponent {
-  componentDidMount() {
-    if (this.props.isModal) document.body.classList.add('modal-open');
-  }
+const ActionPanel = React.forwardRef((props, ref) => {
```
Comment (lteacher) [inline src/components/ActionPanel/index.jsx]: We do seem to be dropping the `PureComponent` off from a few components, not sure if anyone has thoughts on that

Comment (veegandhi) [reply src/components/ActionPanel/index.jsx]: This one should have been dropped as the props has `children` in it. Was redundant in first place

**Pull request merged**